components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -2222,6 +2172,92 @@ private String handlePreConsent(OAuth2Parameters oauth2Params, AuthenticatedUser
         return additionalQueryParam;
     }
 
+    /**
+     * Filter requested claims based on OIDC claims and return the claims which includes in OIDC.
+     *
+     * @param claimsForApproval Consent required claims.
+     * @param oauth2Params OAuth parameters.
+     * @param spTenantDomain Tenant domain.
+     * @return List of claims.
+     * @throws RequestObjectException
+     * @throws ClaimMetadataException
```

Comment:
Shall we add description to exceptions as well, like: "If an error occurred while..."

Reply:
Fixed